akka-actor/src/main/scala/akka/io/dns/AsyncDnsResolver.scala

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import akka.actor.{ Actor, ActorLogging }
+import akka.io.SimpleDnsCache
+import akka.io.dns.protocol.DnsRecordType
+import com.typesafe.config.Config
+import akka.pattern.pipe
+
+import scala.concurrent.Future
+
+final class AsyncDnsResolver(cache: SimpleDnsCache, config: Config) extends Actor with ActorLogging {
+  implicit val ec = context.dispatcher
+
+  override def receive: Receive = {
+    case m: DnsProtocol.Protocol ⇒ onReceive(m)
+  }
+
+  def onReceive(m: DnsProtocol.Protocol): Unit = m match {
+    case DnsProtocol.Resolve(name, mode) ⇒
+      resolve(name, mode) pipeTo sender()
+  }
+
+  def resolve(str: String, types: Set[DnsRecordType]): Future[DnsProtocol.Resolved] = {
+    ???
+  }
```

Comment:
TODO actual logic impl... It can be largely based on akka-dns however it should be cleanedup quite a bit I think